app.vue

```diff
@@ -1,21 +1,48 @@
 <script lang="ts" setup>
 import { useAuth0 } from "@auth0/auth0-vue";
 
+const { $appInsights } = useNuxtApp();
+
 const { isLoading, loginWithRedirect, isAuthenticated, error, logout } =
   useAuth0();
 
 watch(
   isLoading,
   async (loading) => {
     if (loading) return;
+
     if (!isAuthenticated.value) {
+      $appInsights.event("auth0 - redirect to login", {});
       await loginWithRedirect();
     }
   },
   { immediate: true },
 );
 
+watch(error, async () => {
+  const e = error.value;
+  if (e) {
+    const errorCode = (e as any).error;
```

Comment:
I wouldn't cast this to `any` but rather check if it has the properties we expect or whether it is an instance of the class we expect.

Reply:
done